File: trunk/src/pann/Core/Storage.cpp
//Storage.cpp

#include "Storage.h"

namespace pann
{
    namespace Storage
    {
        void autoload(Net& _obj, std::string _filename)
        {
            try {
                std::cout<<"Trying XML...\n";
                load<xml_in>(_obj, _filename);
                return;
            } catch(...) {
            }

            try {
                std::cout<<"Trying TXT...\n";
                load<txt_in>(_obj, _filename);
                return;
            } catch(...) {
            }

            try {
                std::cout<<"Trying BIN...\n";
                load<bin_in>(_obj, _filename);
                return;
            } catch(...) {
            }

            throw E<Exception::FilesystemError>()<<"Storage::autoload(): Can't load "<<_filename<<"\n";
        } //load
    }; //Storage
}; //pann

File: pypann/Core/Storage.py
#
# Storage
#

import cPickle

def load(filename):
    file = open(filename, "rb")
    with file:
        return cPickle.load(file)

def save(obj, filename):
    file = open(filename, "wb")
    with file:
        cPickle.dump(obj, file)

